lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2508,6 +2516,116 @@ signTransaction tl preferredLatestEra keyLookup (rootKey, rootPwd) utxo =
             keyLookup
             inputResolver
 
+type MakeRewardAccountBuilder k =
+    (k 'RootK XPrv, Passphrase "encryption") -> (XPrv, Passphrase "encryption")
+
+-- | Produce witnesses and construct a transaction from a given selection.
+--
+-- Requires the encryption passphrase in order to decrypt the root private key.
+-- Note that this doesn't broadcast the transaction to the network. In order to
+-- do so, use 'submitTx'.
+--
+buildAndSignTransactionNoSelection
+    :: forall (n :: NetworkDiscriminant)
+     . (Typeable n, DelegationAddress n ShelleyKey 'CredFromKeyK)
+    => Tracer IO WalletLog
+    -> TimeInterpreter (Either PastHorizonException)
+    -> DBLayer IO (SeqState n ShelleyKey) ShelleyKey
+    -> NetworkLayer IO Block
+    -> TransactionLayer ShelleyKey 'CredFromKeyK SealedTx
+    -> WalletId
+    -> AnyRecentEra
+    -> Passphrase "user"
+    -> TransactionCtx
+    -> IO (Tx, TxMeta, SealedTx)
+buildAndSignTransactionNoSelection
```

Comment:
I got a little confused by this name as it's the variant called buildAndSignTransaction**NoSelection** which **does** perform coin-selection 🤔 But we should have greater liberties in naming after we remove the original `buildAndSignTransaction` function so 👍

Reply:
You're right, `NoSelection` is a subject for different interpretations here. 🤔  My idea was "build and sign tx (no need to supply a selection)" . The best name is `buildAndSignTransaction` but its already taken. Maybe `buildAndSignTransaction'` is not a terrible idea here, assuming that prime will go away eventually.

Another idea: `buildBalanceSignTransaction` ?

Reply:
> Another idea: buildBalanceSignTransaction ?

`buildBalanceSignTransaction` vs `buildAndSignTransaction` make the similarities less apparent, and possibly suggests false differences (they don't differ in whether the result is balanced), so I'd just go with `buildAndSignTransaction'` now.

Reply:
Agreed to name it `buildAndSignTransactionNew` temporarily.